src/Common/PageCache.cpp

```diff
@@ -172,7 +173,22 @@ void PageCache::autoResize(Int64 memory_usage_signed, size_t memory_limit)
 
     size_t reduced_limit = size_t(memory_limit * (1. - std::min(free_memory_ratio, 1.)));
     size_t target_size = reduced_limit - std::min(peak, reduced_limit);
-    target_size = std::clamp(target_size, min_size_in_bytes, max_size_in_bytes);
+
+    /// if page_cache_max_size setting is for some reason less than page_cache_min_size,
+    /// then set the target_size to page_cache_min_size to ensure that the resizing logic
+    /// works correctly and doesn't cause the server to crash.
+    if (min_size_in_bytes <= max_size_in_bytes)
+    {
+        target_size = std::clamp(target_size, min_size_in_bytes, max_size_in_bytes);
+    }
+    else
+    {
+        target_size = min_size_in_bytes;
+        LOG_WARNING(
```

Comment:
It's better to throw an exception and prevent server startup.

Reply:
Sure, moved the validation to server startup.